pkg/nsx/util/license.go

```diff
@@ -0,0 +1,87 @@
+package util
+
+import (
+	"strconv"
+	"strings"
+	"sync"
+)
+
+const (
+	FeatureContainer         = "CONTAINER"
+	FeatureDFW               = "DFW"
+	LicenseContainerNetwork  = "CONTAINER_NETWORK"
+	LicenseContainerSecurity = "DFW"
+	LicenseContainer         = "CONTAINER"
+)
+
+var (
+	container_network  = false
+	container_security = false
+	licenseMutex       sync.Mutex
+	licenseMap         = map[string]bool{FeatureContainer: container_network, FeatureDFW: container_security}
+)
+
+type NsxLicense struct {
+	Results []struct {
+		FeatureName string `json:"feature_name"`
+		IsLicensed  bool   `json:"is_licensed"`
+	} `json:"results"`
+	ResultCount int `json:"result_count"`
+}
+
+func IsLicensed(feature string) bool {
+	licenseMutex.Lock()
+	defer licenseMutex.Unlock()
+	return licenseMap[feature]
+}
+
+func UpdateLicense(feature string, isLicensed bool) {
+	licenseMutex.Lock()
+	licenseMap[feature] = isLicensed
+	licenseMutex.Unlock()
+}
+
+func searchLicense(licenses *NsxLicense, licenseName string) bool {
+	license := false
+	for _, feature := range licenses.Results {
+		if feature.FeatureName == licenseName {
+			license = feature.IsLicensed
+		}
+	}
+	return license
+}
+func getContainerLicenseName(version string) string {
+	// TODO, change to LicenseContainerNetwork when nsx 4.2 patch merged
```

Comment:
same here, golang has version handing lib, we can avoid splitting versions and use lib instead

Reply:
remove version checking